product_configurator/models/product.py

```diff
@@ -436,24 +438,36 @@ def values_available(self, attr_val_ids, sel_val_ids):
             avail = self.validate_domains_against_sels(domains, sel_val_ids)
             if avail:
                 avail_val_ids.append(attr_val_id)
-
+            elif do_raise:
+                if len(config_lines) == 1 and config_lines[0].rule_description:
```

Comment:
Our use customer's use case is:

They configures the product on a single view, with around 15 options.  Some of the rules and dependencies were too complicated to reflect in readonly's on the view (the other PR will fix a lot of that).  Then they clicked submit and they just got a "configuration is invalid - please check all the steps" error, which did not help them focus on the problem.

The real benefit of this approach is where a configurable product has been defined badly.  A value on step 3 may make a value on step 1 invalid!  This is a bad setup.  But, the rendering of the page cannot foresee this (and nor should it have to).  If they save the configuration after step 3 in this scenario, it just says "please check all your steps", but with a focussed error message, they will see ("Value X should not be chosen when the product is Blue or Red")....